src/Symfony/Component/Console/Completion/CompletionInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Completion;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+interface CompletionInterface
```

Comment:
Is the contract needed?

Reply:
Strictly, no (given that it's implemented by both main contracts of this component; Application and Command). But I thought it would be nice to make sure both application and command use the same completion contract.

Reply:
I'd keep it an implementation detail for now. For example, can we also think of a way `parent::complete` isnt needed?

Either way ... :+1:  :)

Reply:
I've slightly updated the PR based on your suggestion:

- I've moved the input option completion to the `CompleteCommand` directly (removing the parent call)
- `Command` now longer implements `complete()`